### PR TITLE
feat: add reNFT Sylvester adapter

### PR DESCRIPTION
**Pull request by seendsouza**
This PR adds @re-nft collateral-free rental support on Polygon and Ethereum

### CODE REVIEW

File: src/adapters/renft.adapter.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      assetName: name,
+    });
+  }
+
+  async getStakedTokenIds(
+    owner: string,
+    assetName?: string,
+  ): Promise<BigNumber[]> {
+    // Assume assetName is the CAIP asset name
+    const splitAssetName = assetName?.split(':');
```
Comment (raymondfeng) [inline src/adapters/renft.adapter.ts:69]: You can use `new AssetName(assetName)`.

File: src/adapters/renft.adapter.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    const nftAddress =
+      splitAssetName && splitAssetName.length == 2
+        ? splitAssetName[1]
+        : undefined;
+    const query = getSylvesterQuery(owner, nftAddress);
+    if (this.queryUrl === undefined)
+      throw new Error(
+        `ReNFT subgraph URL undefined for ${this.constructor.name}.`,
+      );
+    const body = await fetch(this.queryUrl, {
```
Comment (raymondfeng) [inline src/adapters/renft.adapter.ts:79]: Please use the following:

```ts
import {getFetch, handleFetchResponse} from `@collabland/common`;

const fetch = getFetch();
const res = await fetch(...);
const json = await handleFetchResponse(res);
```

File: src/adapters/renft.adapter.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class ReNFTEthereumSylvesterV0StakingContractAdapter extends BaseReNFTSylvesterStakingContractAdapter {
+  chainId = 1;
+  contractAddress = '0xa8D3F65b6E2922fED1430b77aC2b557e1fa8DA4a';
+  queryUrl = getEnvVar('RENFT_ETHEREUM_SYLVESTER_V0_SUBGRAPH_URL');
```
Comment (raymondfeng) [inline src/adapters/renft.adapter.ts:99]: Do we have a default value for the url?

File: src/adapters/renft.adapter.ts [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+@injectable(
+  {
+    scope: BindingScope.SINGLETON,
+  },
+  extensionFor(STAKING_ADAPTERS_EXTENSION_POINT),
+)
+export class ReNFTPolygonSylvesterV1StakingContractAdapter extends BaseReNFTSylvesterStakingContractAdapter {
+  chainId = 137;
+  contractAddress = '0x4e52b73aa28b7ff84d88ea3a90c0668f46043450';
+  queryUrl = getEnvVar('RENFT_POLYGON_SYLVESTER_V1_SUBGRAPH_URL');
```
Comment (raymondfeng) [inline src/adapters/renft.adapter.ts:111]: Default value?

File: src/adapters/renft.adapter.ts (hunk 3 above)
Comment (seendsouza) [reply src/adapters/renft.adapter.ts]: Added

File: src/adapters/renft.adapter.ts (hunk 4 above)
Comment (seendsouza) [reply src/adapters/renft.adapter.ts]: Added

File: src/adapters/renft.adapter.ts (hunk 2 above)
Comment (seendsouza) [reply src/adapters/renft.adapter.ts]: Fixed

File: src/adapters/renft.adapter.ts (hunk 1 above)
Comment (seendsouza) [reply src/adapters/renft.adapter.ts]: Fixed

**Comment by raymondfeng** [comment]
@seendsouza Can you grant me permission to `seendsouza:master` so that I can push some changes to the PR?

**Comment by seendsouza** [comment]
@raymondfeng invited you

**Pull request merged**